test/unit/util/httpProvider.browser-test.js

```diff
@@ -0,0 +1,532 @@
+    });
+  });
+  test('test format response text when error', () => {
+    const response = HttpProvider.formatResponseText(null);
+    expect(response).toEqual(null);
+  });
+  test('test timeout', async () => {
+    const p = HttpProvider.timeoutPromise(3000);
+    expect(setTimeout).toHaveBeenCalledTimes(1);
+    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 3000);
+    jest.runAllTimers();
+    return expect(p).resolves.toEqual({ type: 'timeout' });
+  });
+  test('test get request send by fetch', async () => {
+    const httpProvider = new HttpProvider(stageEndpoint);
+    const response = await httpProvider.requestSendByFetch(
+      {
+        url: 'block/blockInfo',
+        method: 'GET',
+        params: {
+          height: 43143609,
+        },
+      },
+      fetch
+    );
+    const result = await response.text();
+    expect(JSON.parse(result)).toEqual({
+      id: 43735856,
+      block_hash:
+        '28ab301869791fe34b20747e77d18b364997d647ebd13acd9337a7359c92df06',
+      pre_block_hash:
+        '2270f9afe57f1ab3e99277c0a177f42ac9690de6eb4dc049e5f921ed740703af',
+      chain_id: 'tDVW',
+      dividends: '{}',
+      miner: 'szJDnSZaPBesgzD9ZS3EFaRyUbj2z3YPTzXtcW6MX27AvF4CM',
+      tx_fee: '{}',
+      resources: '{}',
+      block_height: 43143609,
+      tx_count: 2,
+      merkle_root_tx:
+        'e4edc1c356a120027d817b801d1f191fabe0a23188611ca6d38f884984e122fc',
+      merkle_root_state:
+        'd05386093ff74cde09c6144b8e5acae824710a103c1333dd797c088841b470f5',
+      time: '2023-02-21T06:25:49.1205547Z',
+    });
+  });
+  test('test post request send by fetch', async () => {
+    const httpProvider = new HttpProvider('https://aelf-public-node.aelf.io');
+    const response = await httpProvider.requestSendByFetch(
+      {
+        url: 'blockChain/calculateTransactionFee',
+        params: {
+          RawTransaction:
+            '0a220a2005c3b3959caeee55b5db4004f6f9d76860aae818ce7b33d210a446ecb275468212220a200e9a238616169860ba4fb502f8e87ce62db9ef321f29fd76d6e1ce30a0e0cdba18a8a19e022204fa60011c2a085472616e73666572322e0a220a200e9a238616169860ba4fb502f8e87ce62db9ef321f29fd76d6e1ce30a0e0cdba1203454c461880c2d72f82f10441ad1647aa462a8859c7a699f22d6124c82a5239c90746ac67ebfd8e1a7b4a25f343733351162d205fc881ea62072e9fe0fad5f6113734309b6666f6812d6b7a6400',
+        },
+      },
+      fetch
+    );
+    const result = await response.text();
+    expect(JSON.parse(result)).toEqual({
+      Success: false,
+      TransactionFee: null,
+      ResourceFee: null,
+    });
+  });
+  test('test send async by fetch', async () => {
+    const xhr = window.XMLHttpRequest;
+    delete window.XMLHttpRequest;
+    window.fetch = fetch;
+    const NewHttpProvider = require('../../../src/util/httpProvider').default;
+    const httpProvider = new NewHttpProvider(stageEndpoint);
+    window.XMLHttpRequest = xhr;
+    const result = await httpProvider.sendAsyncByFetch({
+      url: 'nodes/info',
```

Comment:
why test api of aelf explore? Please use https://aelf-public-node.aelf.io/swagger/index.html instead  of it.